python-sdk/src/astro/sql/operators/base_decorator.py

```diff
@@ -56,6 +62,31 @@ def __init__(
             **kwargs_with_datasets(kwargs=kwargs, output_datasets=self.output_table),
         )
 
+    def render_template_fields(
+        self,
+        context: Context,
+        jinja_env: jinja2.Environment | None = None,
+    ) -> BaseOperator | None:
+        """Template all attributes listed in template_fields.
+
+        This mutates the attributes in-place and is irreversible.
+
+        :param context: Dict with values to apply on content
+        :param jinja_env: Jinja environment
+        """
+        if not jinja_env:
+            jinja_env = self.get_template_env()
+
+        # We don't resolve undefined variables such as input tables
+        # as this would require opening a connection to the database.
+        # E.g. calling `create_database(self.conn_id)`` etc.
+        jinja_env.undefined = DebugUndefined
+
+        self._do_render_template_fields(
+            self, self.template_fields, context, jinja_env, set()
+        )
+        return self
```

Comment:
Can you explain a bit more on why you're overriding this?

Reply:
Of course :) We are overriding this because the context is not aware of our astro-sdk input tables in jinja. It cannot resolve the `{{ table_1 }}` variables. It raises an `Undefined` error, because we would have to update the context with our input tables. The problem is that we cannot easily update it with that, because this would require retrieving connections from the database and I am not sure if we want to hit the database every time we render the jinja.

We can also take this offline and I show it to you in a demo.